File: DSA/c++/1-Arrays/1-Maximum_and_Minimum_Element_in_an_Array.cpp
#include <iostream>
#include <limits>

void get_min_max(int array[] , int size ,int &min, int &max ){

    min = std::numeric_limits<int>::max();
    max = std::numeric_limits<int>::min();

    for (int i=0; i < size; i++){
        if(array[i]< min){
            min = array[i];
        }
        if(array[i]> max){
            max = array[i];
        }
    }
}

int main(){
    int array1[] = {3, 5, 4, 1, 9};
    int array2[] = {22, 14, 8, 17, 35, 3};
    int min , max;

    get_min_max(array1,5,min , max);
    printf("Array 1: Min = %d, Max = %d\n", min, max);

    get_min_max(array2, 6, min, max);
    printf("Array 2: Min = %d, Max = %d\n", min, max);
    
}

File: DSA/javascript/1-Arrays/1-Maximum_and_Minimum_Element_in_an_Array.js
function getMinMax(array){
    let min = Number.MAX_SAFE_INTEGER;
    let max = Number.MIN_SAFE_INTEGER;

    for (let i=0;i<array.length;i++){
        if(array[i]<min){
            min = array[i];
        }
        if(array[i]>max){
            max = array[i];
        }

    }

return {min : min, max: max};

}

let array1 = [3, 5, 4, 1, 9];
let array2 = [22, 14, 8, 17, 35, 3];

let result1 = getMinMax(array1);
console.log("Array 1: Min =", result1.min, ", Max =", result1.max);

let result2 = getMinMax(array2);
console.log("Array 2: Min =", result2.min, ", Max =", result2.max);